### PR TITLE
Let prow display gubernator url if pod is not found

### PR DESCRIPTION
**Pull request by krzyzacy**
Also make periodic job clickable (link to gubernator url)
ref https://github.com/kubernetes/test-infra/issues/2757

/assign @spxtr @rmmh

### CODE REVIEW

File: prow/cmd/deck/static/script.js [hunk 1]
```diff
@@ -138,6 +138,8 @@ function redraw() {
         r.appendChild(stateCell(build.state));
         if (build.pod_name) {
             r.appendChild(createLinkCell("\u2261", "log?pod=" + build.pod_name));
+        } else if (build.url !== "") {
+            r.appendChild(createLinkCell("\u2261", build.url));
```
Comment (rmmh) [inline prow/cmd/deck/static/script.js]: this is wrong, the link cells should only point to locally-hosted logs

File: prow/cmd/deck/static/script.js [hunk 2]
```diff
@@ -167,7 +169,7 @@ function redraw() {
             r.appendChild(createTextCell(""));
             r.appendChild(createTextCell(""));
         }
-        if (build.url === "" || build.type === "periodic") {
+        if (build.url === "") {
```
Comment (rmmh) [inline prow/cmd/deck/static/script.js]: I don't think any of the periodic jobs have correct build.urls currently. Let's fix that first! One has:

    "url": "https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/_/ci-kubernetes-e2e-gci-gke-reboot-release-1-7/338/",

File: prow/cmd/deck/static/script.js (hunk 1 above)
Comment (krzyzacy) [inline prow/cmd/deck/static/script.js]: it's used below -
`r.appendChild(createLinkCell(build.job, build.url));`

Comment (rmmh) [inline prow/cmd/deck/static/script.js]: sorry, I mean that this column should only point to locally-hosted logs.

Comment (krzyzacy) [inline prow/cmd/deck/static/script.js]: emmm, the goal is to have some log place for finished pods. Maybe the below line is enough, or I can use a different icon here.

File: prow/cmd/deck/static/script.js (hunk 2 above)
Comment (krzyzacy) [inline prow/cmd/deck/static/script.js]: added fix to plank

File: prow/plank/controller.go [hunk 3]
```diff
@@ -412,14 +412,15 @@ func guberURL(pj kube.ProwJob, build string) string {
 	var url string
 	if pj.Spec.Type == kube.PresubmitJob || pj.Spec.Type == kube.BatchJob {
 		url = guberBasePR
+		if pj.Spec.Refs.Org != "kubernetes" {
```
Comment (rmmh) [inline prow/plank/controller.go]: This is wrong. we still need this logic for push jobs. 

Can you add controller_test.go to exercise the different cases of this function?

Comment (krzyzacy) [inline prow/plank/controller.go]: periodic job does not have org/repo specified though, I'll do it the right way

Comment (rmmh) [inline prow/plank/controller.go]: seems like we should make periodic jobs tied to an org/repo, especially since we're migrating jobs from jenkins

Comment (krzyzacy) [inline prow/plank/controller.go]: org is reasonable, but it does not tie to a repo though (can be none or multiple)

File: prow/cmd/deck/static/script.js (hunk 1 above)
Comment (krzyzacy) [inline prow/cmd/deck/static/script.js]: changed that to 'G' so it can better indicate gubernator

**Comment by krzyzacy** [comment]
added tests as well

File: prow/plank/controller.go (hunk 3 above)
Comment (spxtr) [inline prow/plank/controller.go]: Periodic jobs are tied to neither org nor repo.

**Comment by spxtr** [comment]
Do the fixed links actually point to the right place for periodic jobs?

**Comment by krzyzacy** [comment]
@spxtr yeah, previously it appends _ because it does not have org/repo and the org is not kubernetes.

**Comment by spxtr** [comment]
/lgtm

Please bump plank and deck versions.

One thing to note is that this is very soon going to be templated away. We're going to have to stop special-casing kubernetes, which I think means we'll have to upload to the full path in bootstrap.py and also look there in gubernator.

File: prow/cmd/deck/static/script.js [hunk 4]
```diff
@@ -138,6 +138,8 @@ function redraw() {
         r.appendChild(stateCell(build.state));
         if (build.pod_name) {
             r.appendChild(createLinkCell("\u2261", "log?pod=" + build.pod_name));
+        } else if (build.url !== "") {
+            r.appendChild(createLinkCell("G", build.url));
```
Comment (spxtr) [inline prow/cmd/deck/static/script.js]: Whoops, I didn't notice this when I reviewed. This seems wrong, we now have two links to `build.url` which is strange. Plus, the G probably stands for Gubernator, but doesn't make sense if the link doesn't actually go to Gubernator.

Comment (krzyzacy) [inline prow/cmd/deck/static/script.js]: ah, guess the intention was to not have the area blank. I'll get rid of it then

Comment (krzyzacy) [inline prow/cmd/deck/static/script.js]: https://github.com/kubernetes/test-infra/pull/3171

**Pull request merged**